ark/spLDA/processing.py

```diff
@@ -63,24 +52,153 @@ def format_cell_table(cell_table, markers=None, clusters=None, fovs="all_fovs"):
         })
 
     # Create dictionary of FOVs
-    if fovs == "all_fovs":
-        fovs = [x + 1 for x in range(max(cell_table["point"]))]
+    fovs = np.unique(cell_table["point"])
 
     fov_dict = {}
     for i in fovs:
-        df = cell_table[cell_table["point"] == i].drop(columns=["point", "label"])
+        df = cell_table[cell_table["point"] == i].drop(
+            columns=["point", "label"])
         if clusters is not None:
             df = df[df["cluster_id"].isin(clusters)]
-        df['is_index'] = True
-        df['isimmune'] = True
+        df["is_index"] = True
+        df["is_immune"] = True  # might remove this
         fov_dict[i] = df.reset_index(drop=True)
 
-    return(fov_dict)
+    # Save Arguments
+    fov_dict["fovs"] = fovs
+    fov_dict["markers"] = markers
+    fov_dict["clusters"] = clusters
+
+    return fov_dict
 
 
+def featurize_cell_table(cell_table, feature_by="cluster", radius=100,
+                         cell_index="is_index", n_processes=None,
+                         train_split=True):
+    """Calculates statistics for local cellular neighborhoods based on the
+    specified features and radius.
 
+    Args:
+        cell_table (dict, pd.DataFrame):
+            A formatted cell table for use in spatial-LDA analysis.
+            Specifically, this is the output from
+            format_cell_table().
+        feature_by (str):
+            One of four choices of featurization method:
+                * marker: for each marker, count the total number of cells
+                within a ``radius`` *r* from cell *i*
+                having marker expression greater than 0.5.
+                * avg_marker: for each marker, compute the average marker
+                expression of all cells with a ``radius`` *r*
+                from cell *i*.
+                * cluster: for each cluster, count the total number of cells
+                with a ``radius`` *r* from cell *i*
+                belonging to that cell cluster.
+                * count: counts the total number of cells within a ``radius``
+                *r* from cell *i*.
+        radius (int):
+            Size of the radius, in pixels, used to featurize cellular
+            neighborhoods.
+        cell_index (str):
```

Comment:
If we default to this, do we need to add index in `format_cell_table`?

Reply:
I'm not sure what you mean.

Reply:
Is this line necessary if we have this option enabled? 
![image](https://user-images.githubusercontent.com/13770365/128050996-d7d313b9-8551-49be-8a18-b693edda3f0f.png)

Reply:
Meaning if cell_index is none, it assumes all cells are part of the index. So we can have cell_index be none, and not need to add a cell_index column to the data_table?

Reply:
The featurize_samples() function needs a reference column in each FOV which it uses to run the featurization step.  In this case, "is_index" tells that function that all cells are to be used.  Alternatively, the user could specify any other existing column name that has values equal True for reference cells and False for non-reference cells, but they would have had to create this column on their own beforehand.  So either way a reference column has to exist.

Reply:
kk, sounds good